static/app/components/profiling/transactionProfileIdProvider.tsx

```diff
@@ -0,0 +1,56 @@
+import {createContext, useContext, useEffect} from 'react';
+import * as Sentry from '@sentry/react';
+
+import {useProfileEvents} from 'sentry/utils/profiling/hooks/useProfileEvents';
+
+const TransactionProfileContext = createContext<string | null | undefined>(undefined);
+
+interface TransactionToProfileIdProviderProps {
+  children: React.ReactNode;
+  transactionId: string | undefined;
+}
+
+export function TransactionProfileIdProvider({
+  transactionId,
+  children,
+}: TransactionToProfileIdProviderProps) {
+  const {status, data, error} = useProfileEvents({
+    fields: ['id'],
+    referrer: 'transactionToProfileProvider',
+    limit: 1,
+    sort: {
+      key: 'id',
+      order: 'asc',
+    },
+    query: `trace.transaction:${transactionId}`,
+    enabled: Boolean(transactionId),
+  });
```

Comment:
We talked on slack but I think there's more to this. If you shared a link to a transaction without any of the query parameters, the default page filter will get used. This results in a stats period of 14 days I believe. So if the event shared is older than that, then this query will never find the profile. The fix here is likely what I mentioned and restrict the time window the transaction event's timestamp +/- 12 hours (this is pretty arbitrary but a 24 hour window around the event should give us nearly 100% success rate.